internal/clientcache/internal/client/client.go

```diff
@@ -0,0 +1,150 @@
+package client
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/hashicorp/boundary/api"
+	"github.com/hashicorp/boundary/internal/clientcache/internal/daemon"
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/version"
+	"github.com/hashicorp/go-cleanhttp"
+	"github.com/hashicorp/go-retryablehttp"
+)
+
+const hostHeader = "api.boundary.localhost"
+
+type Client struct {
+	client *retryablehttp.Client
+	u      *url.URL
+}
+
+// New returns a client which will always send requests over the unix socket
+// specified in the provided address.  If the address doesn't have the "unix"
+// schema or the path is unset, an error is returned.
+func New(ctx context.Context, address *url.URL) (*Client, error) {
+	const op = "client.New"
+	switch {
+	case address == nil:
+		return nil, errors.New(ctx, errors.InvalidParameter, op, "address is nil")
+	case address.Scheme != "unix":
+		return nil, errors.New(ctx, errors.InvalidParameter, op, "address does not have a unix schema")
+	case address.Path == "":
+		return nil, errors.New(ctx, errors.InvalidParameter, op, "address path is empty")
+	}
+	c := &retryablehttp.Client{
+		HTTPClient:   cleanhttp.DefaultClient(),
+		RetryWaitMin: 100 * time.Millisecond,
+		RetryWaitMax: 1500 * time.Millisecond,
+		RetryMax:     6,
+		CheckRetry:   retryablehttp.DefaultRetryPolicy,
+		Backoff:      retryablehttp.DefaultBackoff,
+		ErrorHandler: retryablehttp.PassthroughErrorHandler,
+	}
+	transport := c.HTTPClient.Transport.(*http.Transport)
+	transport.DialContext = func(ctx context.Context, n string, a string) (net.Conn, error) {
+		dialer := net.Dialer{}
+		return dialer.DialContext(ctx, "unix", address.Path)
+	}
+	return &Client{client: c, u: address}, nil
+}
+
+// Response returns the *http.Response as well as providing the body of the response
+// through a call to Body().
+type Response struct {
+	*http.Response
+	contents []byte
+}
+
+// Body returns the pre-read response body.
+func (r Response) Body() []byte {
+	return r.contents
+}
+
+// Get sends a GET http request to the provided path.  The vals provided are
+// encoded and attached to the request if present.
+func (c *Client) Get(ctx context.Context, path string, vals *url.Values) (*Response, *api.Error, error) {
```

Comment:
If you wanted you could probably combine these, or share an internal function between GET/POST. Either way works.